### ISSUE TITLE
ng build --watch build all bundles instead delta bundle

### ISSUE THREAD

**Issue by almogyehuda** [angular/angular-cli#18660]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [x ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

Yes

ng build --watch should produce only delta bundle, and not build the whole project on any change.

### Description

when using ng build --watch flag, I expect the behavior to be as same as it was in previous version:

If the main code changed, the build --watch should produce only new main.js bundle, if the styles changed, should
produce new styles.js bundle.

Actual - the ng build --watch produce all bundles in each change (change on main/styles etc.)

## 🌍 Your Environment
<pre><code>
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/

Angular CLI: 10.0.6
Node: 10.14.2
OS: win32 x64

Angular: 10.0.6
... cli, core, forms, platform-browser, platform-browser-dynamic
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.1000.6
@angular-devkit/build-angular     0.1000.6
@angular-devkit/build-optimizer   0.1000.6
@angular-devkit/build-webpack     0.1000.6
@angular-devkit/core              8.0.0
@angular-devkit/schematics        8.0.0
@angular/cdk                      7.3.7
@angular/common                   10.0.11
@angular/compiler                 10.0.11
@angular/compiler-cli             10.0.11
@angular/http                     7.2.16
@angular/router                   10.0.8
@ngtools/webpack                  10.0.6
@schematics/angular               8.0.0
@schematics/update                0.1000.6
rxjs                              6.6.2
typescript                        3.9.7
webpack                           4.39.2
</code></pre>

**Comment by alan-agius4** [comment]
Hi @almogyehuda,  I'm sorry, but we can't reproduce the problem following the instructions you provided.
Remember that we have a large number of issues to resolve, and have only a limited amount of time to reproduce your issue.

Short, explicit instructions make it much more likely we'll be able to reproduce the problem so we can fix it.

A good way to make a minimal repro is to create a new app via `ng new repro-app` and add the minimum possible code to show the problem. Then you can push this repository to github and link it here.

**Comment by almogyehuda** [comment]
> Hi @almogyehuda, I'm sorry, but we can't reproduce the problem following the instructions you provided.
> Remember that we have a large number of issues to resolve, and have only a limited amount of time to reproduce your issue.
> 
> Short, explicit instructions make it much more likely we'll be able to reproduce the problem so we can fix it.
> 
> A good way to make a minimal repro is to create a new app via `ng new repro-app` and add the minimum possible code to show the problem. Then you can push this repository to github and link it here.

Hey,
Invite you as collaborator to the minimal repro.

Please run npm run build:dev:watch and make some change (change on the .scss file or something)
You will see that all the bundles built again

Thanks.

**Comment by almogyehuda** [comment]
While playing with the repro, I found that when I set "optimization" to false, the watch is working as expected.
But why is that? @alan-agius4 

Thanks.

**Comment by alan-agius4** [comment]
Hi @almogyehuda,

I looked at your reproduction and this behaviour is because because for your development build you are setting `optimization` to `true`.  When optimization is set to true, Webpack will also run in `production` mode (https://webpack.js.org/configuration/mode/#mode-production).

This will enable more plugins for better code splitting and dead code elimination. This does however comes with a performance cost and it's not recommended to use `production` together with `watch` mode. One of the drawbacks is that for every change each bundle will need to be optimized and re-emit. This is because any change in any bundle can effect the other bundles when Webpack is run in production mode.

Closing as this is working as expected as this is how Webpack handles `production` mode.
